storage-proofs/src/zigzag_graph.rs

```diff
@@ -42,6 +71,20 @@ where
         expansion_degree: usize,
         seed: [u32; 7],
     ) -> Self {
+        let cache_entries = match get_config("MAXIMIZE_CACHING") {
+            Result::Ok(config) => {
+                if config {
+                    info!(SP_LOG, "using parents cache of unlimited size",);
+                    nodes
+                } else {
+                    0
+                }
+            }
+            Result::Err(_) => 0,
+            // If we can't find the `MAXIMIZE_CACHING` assume the conservative
+            // option of no cache.
+        };
+
```

Comment:
This looks good for now. We will probably move this logic into another layer later, but putting it at the point of use seems optimal for present purposes.